Zend/zend_execute.c

```diff
@@ -675,68 +675,59 @@ static ZEND_COLD void zend_verify_type_error_common(
 
 	if (is_union_type(arg_info->type)) {
```

Comment:
You should be able to simplify this code by keeping only the part in this branch.